[BUG] New catalogs don't have the same channels and dependencies as the old manual index

# Version

ca8e594

# What happened?

`make test-e2e` fails with published images from new index generation by identifying that there are different numbers of channel heads than previously. Ex:

```
~/Projects/oc-mirror
tar: Removing leading `/' from member names
/configs/index.json
number of bundles mirrored (5) does not match expected number (6)
make: *** [Makefile:51: test-e2e] Error 1
```

# What did you expect to happen?

E2E tests to pass

# How to reproduce it (as minimally and precisely as possible)?

```console
$ make publish-catalog
$ make test-e2e
```

# Anything else we need to know?

The affected images have already been republished.

--- a/test/operator/bundles/bar/bar-bundle-v1.0.0/metadata/annotations.yaml
+++ b/test/operator/bundles/bar/bar-bundle-v1.0.0/metadata/annotations.yaml
@@ -1,4 +1,4 @@
 annotations:
   operators.operatorframework.io.bundle.package.v1: bar
-  operators.operatorframework.io.bundle.channels.v1: stable
+  operators.operatorframework.io.bundle.channels.v1: alpha,stable
   operators.operatorframework.io.bundle.channel.default.v1: stable

--- a/test/operator/bundles/foo/foo-bundle-v0.2.0/metadata/annotations.yaml
+++ b/test/operator/bundles/foo/foo-bundle-v0.2.0/metadata/annotations.yaml
@@ -1,4 +1,4 @@
 annotations:
   operators.operatorframework.io.bundle.package.v1: foo
-  operators.operatorframework.io.bundle.channels.v1: beta,stable
+  operators.operatorframework.io.bundle.channels.v1: beta
   operators.operatorframework.io.bundle.channel.default.v1: beta

--- a/test/operator/bundles/foo/foo-bundle-v0.3.1/metadata/dependencies.yaml
+++ b/test/operator/bundles/foo/foo-bundle-v0.3.1/metadata/dependencies.yaml
@@ -3,7 +3,7 @@ dependencies:
   - type: olm.package
     value:
       packageName: bar
-      version: <0.3.0
+      version: <0.2.0
   - type: olm.gvk
     value:
       group: "test.bar"

--- a/test/operator/publish_images.py
+++ b/test/operator/publish_images.py
@@ -99,7 +99,7 @@ def build_and_push(context: Path = None, image: str = None, extra_args: str = No
     os.chdir(context)
 
     # Build
-    base_command = f'{RUNTIME} build . --format=oci -t {image}'
+    base_command = f'{RUNTIME} build . --format=docker -t {image}'
     if extra_args is None:
         command = base_command
     else:
@@ -108,7 +108,7 @@ def build_and_push(context: Path = None, image: str = None, extra_args: str = No
         logger.debug(line)
 
     # Push
-    for line in shell(f'{RUNTIME} push --format=oci {image}'):
+    for line in shell(f'{RUNTIME} push --format=docker {image}'):
         logger.debug(line)
 
     logger.debug(f'Returning to {prev_dir}')

